src/benchq/data_structures/hardware_architecture_models.py

```diff
@@ -47,3 +50,216 @@ class SCModel:
 
 BASIC_ION_TRAP_ARCHITECTURE_MODEL = IONTrapModel()
 BASIC_SC_ARCHITECTURE_MODEL = SCModel()
+
+
+@dataclass
+class DetailedIonTrapResourceInfo:
+    """Info relating to detailed ion trap architecture model resources."""
+
+    power_consumed_per_elu_in_kW: float
+    num_communication_ports_per_elu: int
+    second_switch_per_elu_necessary: bool
+    num_communication_qubits_per_elu: int
+    num_memory_qubits_per_elu: int
+    num_computational_qubits_per_elu: int
+    num_optical_cross_connect_layers: int
+    num_ELUs_per_optical_cross_connect: int
+
+    total_num_ions: int
+    total_num_communication_qubits: int
+    total_num_memory_qubits: int
+    total_num_computational_qubits: int
+    total_num_communication_ports: int
+    num_elus: int
+    total_elu_power_consumed_in_kW: float
+    total_elu_energy_consumed_in_kJ: float
+
+
+class DetailedIonTrapModel:
+    def __init__(
+        self,
+        physical_qubit_error_rate: float = 1e-4,
+        surface_code_cycle_time_in_seconds: float = 1e-3,
+    ):
+        self.physical_qubit_error_rate = 1e-4
+        # TODO: PJ check with Simon about this number
+        self.surface_code_cycle_time_in_seconds = 1e-3
```

Comment:
The arguments `physical_qubit_error_rate` and `surface_code_cycle_time_in_seconds` should be assigned to the corresponding attributes.

Reply:
Done in most recent commit!